File: app/src/main/java/temple/edu/yelp_randomizer/restaraunts/RecycleAdapter.java
package temple.edu.yelp_randomizer.restaraunts;

import android.content.Context;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.recyclerview.widget.RecyclerView;
import com.squareup.picasso.Picasso;
import temple.edu.yelp_randomizer.R;
import temple.edu.yelp_randomizer.View.ReviewsView;
import temple.edu.yelp_randomizer.models.ReviewsModel;

import java.util.ArrayList;

public class RecycleAdapter extends RecyclerView.Adapter<ReviewsView> {
    private ArrayList<ReviewsModel> reviews;
    private Context context;

    public RecycleAdapter(Context context, ArrayList<ReviewsModel> reviews) {
        this.reviews = reviews;
        this.context = context;
    }

    @Override
    public ReviewsView onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.reviews_view, parent, false);
        return new ReviewsView(view);
    }

    @Override
    public void onBindViewHolder(ReviewsView holder, int position) {
        ReviewsModel currentModel = reviews.get(position);
        holder.getReviewTextTextView().setText(currentModel.getText());
        holder.getReviewRatingTextView().setText(convertRating(currentModel.getRating()));
        holder.getReviewNameTextView().setText(currentModel.getUserName());
        if (currentModel.getProfileImageURL() != null) {
            Picasso.with(context).load(currentModel.getProfileImageURL()).into(holder.getProfilePictureImageView());
        }

    }


    @Override
    public int getItemCount() {
        return reviews.size();
    }

    private String convertRating(int rating){
        String rating_stars = "";
        for (int i = 0; i<= rating-1; i++){
            rating_stars += '⭐';
        }
        return rating_stars;
    }
}

File: app/src/main/java/temple/edu/random/restaraunts/RecycleAdapter.kt
package temple.edu.random.restaraunts

import android.content.Context
import android.view.ViewGroup
import temple.edu.random.models.ReviewsModel
import androidx.recyclerview.widget.RecyclerView
import temple.edu.random.View.ReviewsView
import android.view.LayoutInflater
import temple.edu.random.R
import com.squareup.picasso.Picasso
import java.util.ArrayList

class RecycleAdapter(private val context: Context, private val reviews: ArrayList<ReviewsModel>) : RecyclerView.Adapter<ReviewsView>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ReviewsView {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.reviews_view, parent, false)
        return ReviewsView(view)
    }

    override fun onBindViewHolder(holder: ReviewsView, position: Int) {
        val currentModel = reviews[position]
        holder.reviewTextTextView.text = currentModel.text
        holder.reviewRatingTextView.text = convertRating(currentModel.rating)
        holder.reviewNameTextView.text = currentModel.userName
        if (currentModel.profileImageURL != null) {
            Picasso.with(context).load(currentModel.profileImageURL).into(holder.profilePictureImageView)
        }
    }

    override fun getItemCount(): Int {
        return reviews.size
    }

    private fun convertRating(rating: Int): String {
        var rating_stars = ""
        for (i in 0..rating - 1) {
            rating_stars += '⭐'
        }
        return rating_stars
    }
}